src/aap_eda/tasks/ruleset.py

```diff
@@ -286,13 +303,41 @@ def _start_failed(now: timezone.datetime):
             f"Restart activation {activation.name} according to its restart"
             " policy."
         )
-        _schedule_activate(activation, "SCHEDULER")
+        activate(activation.id, "SCHEDULER")
 
 
-def _schedule_activate(activation: models.Activation, requester: str) -> None:
-    job = activate(
-        activation_id=activation.id,
-        requester=requester,
-    )
-    activation.current_job_id = job.id
-    activation.save(update_fields=["current_job_id"])
+def _reconnect_activation_workers() -> list[int]:
+    running_statuses = [
+        ActivationStatus.RUNNING,
+        ActivationStatus.STARTING,
+    ]
+    reconnecting_activations = []
+    for activation in models.Activation.objects.filter(
+        status__in=running_statuses
+    ):
+        if not _worker_alive(activation.worker_name):
+            logger.info(
+                f"The activation worker for actvation_id {activation.id} is "
+                "dead. Will find another worker to reconnect to the instance"
+            )
+            job_id = _activation_job_id(activation.id)
+            job = job_from_queue(get_queue(name="activation"), job_id)
+            if job:
+                job.cancel()
+            activate(activation.id, "SCHEDULER", True)
```

Comment:
Can we use named parameters? It is hard to read specially when are involved booleans that you can not know what mean without look at the function signature